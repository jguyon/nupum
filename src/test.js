import React, { StrictMode } from "react";
import { render as rtlRender, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";
import serializer from "jest-emotion";

expect.addSnapshotSerializer(serializer);

afterEach(() => {
  cleanup();
});

export * from "react-testing-library";

export function render(ui, options) {
  const { rerender: rtlRerender, ...rest } = rtlRender(wrapUi(ui));

  function rerender(nextUi) {
    rtlRerender(wrapUi(nextUi));
  }

  return {
    ...rest,
    rerender,
  };
}

function wrapUi(ui) {
  return <StrictMode>{ui}</StrictMode>;
}
