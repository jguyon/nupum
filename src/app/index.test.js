import React from "react";
import { render } from "../test";
import App from ".";

test("content is rendered", () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
