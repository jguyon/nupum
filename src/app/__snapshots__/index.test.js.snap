// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content is rendered 1`] = `
<DocumentFragment>
  .emotion-1 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  font-family: sans-serif;
  color: pink;
}

<div
    class="emotion-1"
  >
    <h1
      class="emotion-0"
    >
      Hello, world!
    </h1>
  </div>
</DocumentFragment>
`;
